docs/dev.md

```diff
@@ -2,6 +2,50 @@
 
 This document contains information helpful for developers.
 
+## Changelog management
+
+We use [Towncrier](https://towncrier.readthedocs.io) for changelog management. We keep the changelog entries for currently unreleased
+changed in the [.changelog] directory. The contents of this directory are consumed when the changelog is updated prior to a release.
+
+### Adding a changelog entry
+
+If you want to add a changelog entry for your PR, run:
+
+```bash
+make add-changelog-entry
+```
+
+You can also just create the file manually. The filename format is `<PR NUMBER>.<CHANGE TYPE>(.<FRAGMENT NUMBER>).txt`, and the content is
+the entry text.
+
+### My change doesn't need a changelog entry
+
+Add a `Skip Changelog` label to your Pull Request.
+
+### How do I update the changelog while releasing?
+
+```bash
+make update-changelog VERSION=x.x.x
+```
+
+### How do I add multiple entries for the same PR?
+
+Add a counter to your entry file names. For example, if you wanted to have three entries for PR `#123`, which is a fix, you'd create the
+following files in [.changelog]:
+
+```text
+123.fixed.0.txt
+123.fixed.1.txt
+123.fixed.2.txt
+```
+
+Unfortunately, the `towncrier create` command doesn't have support for this [yet](https://github.com/twisted/towncrier/issues/474), so you
+need to create the files manually.
```

Comment:
Should we allow to have multiple `README.md` entries for one PR?

Reply:
Why not? It doesn't come up often, but for PRs that change the support matrix it's often more readable to have multiple change entries for a single PR. See Changelog for 3.1.0 for an example.

I guess we could also have a multiline entry in that case, but I leave that up to the PR author.

Reply:
I was rather thinking that it's not good to have multiple changes in one PR :D